### ISSUE TITLE
Deprecate old backward compatible shims in configure_formatter()/handler().

### ISSUE THREAD

**Issue by felixxm** [python/cpython#115032]
# Feature or enhancement

### Proposal:

`DictConfigurator.configure_formatter()` and `configure_handler()` contain workarounds for old configurations

https://github.com/python/cpython/blob/bcccf1fb63870c1b7f8abe246e27b7fff343abd7/Lib/logging/config.py#L670-L676

https://github.com/python/cpython/blob/bcccf1fb63870c1b7f8abe246e27b7fff343abd7/Lib/logging/config.py#L844-L851

Django doesn't use `fmt` and `strm` for many years. I think both can be deprecated and removed.

I'd like to prepare a patch, if accepted.

### Has this already been discussed elsewhere?

This is a minor feature, which does not need previous discussion elsewhere

### Linked PRs
* gh-115303
* gh-115314

**Comment by sobolevn** [comment]
cc @vsajip

**Comment by vsajip** [comment]
I realise Django doesn't use them any more, but what about other code / users? I wouldn't strongly object to removing them, but there's a chance that any such change might need to be reverted if there are complaints that it breaks existing code somewhere. And, of course, no backporting of any such change. Perhaps deprecate, but not remove immediately - rather, in line with the usual deprecation cycle. It's just a little wart in the current code, after all.

**Comment by felixxm** [comment]
> Perhaps deprecate, but not remove immediately - rather, in line with the usual deprecation cycle.

Yes that's what I was trying to propose. We have an existing and straightforward upgrade path i.e. use `format` instead of `fmt` and use `stream` instead of `strm`, so it seems fine to deprecate the current workarounds and remove them in 3.15 (I guess :thinking:).

> It's just a little wart in the current code, after all.

True, but they become an issue when there are dozens of such workarounds (_"death by a thousand cuts"_).

**Comment by vsajip** [comment]
> become an issue when there are dozens of such workarounds

Well, it's not that bad IMO, but I'll look at a PR that does deprecation.

**Comment by felixxm** [comment]
While preparing a patch, I noticed that we can deprecate `strm`, however, `fmt` is still needed because `config` is passed directly to the custom formatters and [`Formatter`](https://docs.python.org/3/library/logging.html#logging.Formatter) except `fmt` instead of `format`.
